core/chain-configs/src/genesis_config.rs

```diff
@@ -264,6 +266,37 @@ impl GenesisRecords {
     }
 }
 
+pub struct GenesisJsonHasher {
+    digest: sha2::Sha256,
+}
+
+impl GenesisJsonHasher {
+    pub fn new() -> Self {
+        Self { digest: sha2::Sha256::new() }
+    }
+
+    pub fn process_config(&mut self, config: &GenesisConfig) {
+        let mut ser = Serializer::pretty(&mut self.digest);
+        config.serialize(&mut ser).expect("Error serializing the genesis config.");
+    }
+
+    pub fn process_record(&mut self, record: &StateRecord) {
+        let mut ser = Serializer::pretty(&mut self.digest);
```

Comment:
Is there a reason why we are using `pretty` here? My understanding is that we are changing the hash anyway, so we wight as well use the compact form, to hash less bytes.